docs/user_guides/Gufi For Coders/main.tex

```diff
@@ -0,0 +1,226 @@
+\documentclass{article}
+
+% Language setting
+% Replace `english' with e.g. `spanish' to change the document language
+\usepackage[english]{babel}
+
+% Set page size and margins
+% Replace `letterpaper' with`a4paper' for UK/EU standard size
+\usepackage[letterpaper,top=2cm,bottom=2cm,left=3cm,right=3cm,marginparwidth=1.75cm]{geometry}
+
+% Useful packages
+\usepackage{amsmath}
+\usepackage{graphicx}
+\usepackage[colorlinks=true, allcolors=blue]{hyperref}
+
+\title{GUFI: A Coder's Guide}
+\author{Braeden Slade}
+
+\begin{document}
+\maketitle
+
+\section{Introduction}
+Over the years, the amount of data we store and use has grown exponentially to the point that petabytes of storage is not uncommon. What used to be a simple task of accessing and sorting through information has been compounded into an arduous task with the size and scale of super-computing data centers. Being able to query data effectively, while also taking into account permissions becomes paramount into accomplishing daily tasks. This is what the Grand Unified File Index(GUFI) tool aims to accomplish. 
+
+\subsection{Background}
+This is accomplished by recreating the tree structure via indexing. Each directory contains an SQL database file that stores the metadata of the files as well as summary information for that directory and optionally summary information for the entire tree below that directory.
+
+\begin{figure} [h]
+\centering
+\includegraphics[width=0.5\textwidth]{gufi_structure.png}
+\caption{\label{fig:gufi\_structure}Layout and user interaction with GUFI}
+\end{figure}
+
+\clearpage
+
+\section{Understanding the data and how its stored}
+The data stored in these databases include filenames, access and creation dates, file attributes, and extended attributes. Depending on which table is viewed and what view is selected, there can be up to 59 attributes(columns) per entry(row)\\
+\\
+Below are the most commonly used tables and views. 
+
+\begin{figure} [h]
+\centering
+\includegraphics[width=1.0\textwidth]{Database_Schemas.png}
+\caption{\label{fig:Database Schema}Database schema}
+\end{figure}
+
+\subsection{Why pentries is the most commonly used}
+\begin{itemize}
+  \item Provides parent inode as a query-able variable to the entries table
+  \item Exists solely because parent inodes are not stored
+  \item Parent inode is calculated/looked up so that parent inodes are never stored with child records.
+\end{itemize}
+
+\clearpage
+
+\section{gufi\_query}
+
+\subsection{Outline}
+
+gufi\_query is used in order to query information from the generated databases that include all of the information about the indexed root directory and its sub-directories while also taking into account permissions 
+\subsection{Flags}
+
+\begin{table} [h]
+\centering
+\begin{tabular}{l|p{6cm}}
```

Comment:
6cm?

Reply:
This is to ensure that the formatting looks nice when the file is compiled into a pdf

Reply:
Why 6cm and not 5 or 7? Is it even needed?

Reply:
I will send pictures, If I don't use any formatting, the table  is messed up due to how much writing there is. Then I chose 6 as i felt it looked the best. I have images showing 5,6, and 7 for your reference.


![without anything formatting](https://user-images.githubusercontent.com/43049797/178167604-adfacc5c-7f0d-4d66-b873-f1aebc7baafe.png)

![5cm](https://user-images.githubusercontent.com/43049797/178167612-359f26fa-ebfe-4dcc-921e-345e58d68944.png)

![6cm](https://user-images.githubusercontent.com/43049797/178167615-bfa34d5e-029b-455d-8587-a8ca8c51b0bc.png)

![7cm](https://user-images.githubusercontent.com/43049797/178167621-2d8494b2-0839-4bde-bda3-2b08a27fe165.png)

Reply:
try replacing `p{6cm}` with `l`

Reply:
When I do this, the lines are cut off
![cuts off](https://user-images.githubusercontent.com/43049797/178984470-518d6d3a-76ac-4b0f-b2dd-caef2168a1fd.png)